### PR TITLE
feat(DBSS): add dbss flavros data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dbss flavros data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dbss flavros data source
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dbss/ TESTARGS='-run TestAccDatasourceDbssFlavors_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss/ -v -run TestAccDatasourceDbssFlavors_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDbssFlavors_basic
=== PAUSE TestAccDatasourceDbssFlavors_basic
=== CONT  TestAccDatasourceDbssFlavors_basic
--- PASS: TestAccDatasourceDbssFlavors_basic (36.30s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      36.337s
```

### CODE REVIEW

File: huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+			},
+			"level": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the level of the flavor.`,
+			},
+			"memory": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: `Specifies the memory size(MB) in the flavor.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go]: it's better to use GB(1024*1024 MB)

File: huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+func dbssFlavorsFlavorSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the ID of the flavor.`,
+			},
+			"name": {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go]: as id and name are same, I think we can just keep `id`

File: huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+func filterGetFlavorsResponseBodyFlavor(all []interface{}, d *schema.ResourceData) []interface{} {
+	rst := make([]interface{}, 0, len(all))
+	for _, v := range all {
+		if param, ok := d.GetOk("name"); ok &&
+			fmt.Sprint(param) != fmt.Sprint(utils.PathSearch("name", v, nil)) {
+			continue
+		}
+		if param, ok := d.GetOk("availability_zone"); ok {
+			azs := utils.PathSearch("availability_zones", v, make([]interface{}, 0)).([]interface{})
```
Comment (ShiChangkuo) [inline huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go:198]: why convert the default value to `[]interface{}`?

File: huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go (hunk 1 above)
Comment (houpeng80) [reply huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go]: done

File: huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go (hunk 2 above)
Comment (houpeng80) [reply huaweicloud/services/dbss/data_source_huaweicloud_dbss_flavors.go]: fixed

**Comment by houpeng80** [comment]
make testacc TEST=./huaweicloud/services/acceptance/dbss/ TESTARGS='-run TestAccDatasourceDbssFlavors_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dbss/ -v -run TestAccDatasourceDbssFlavors_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceDbssFlavors_basic
=== PAUSE TestAccDatasourceDbssFlavors_basic
=== CONT  TestAccDatasourceDbssFlavors_basic
--- PASS: TestAccDatasourceDbssFlavors_basic (45.89s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dbss      45.921s

**Pull request merged**